posts/build-an-event-sourcing-system-on-aws-using-dynamodb-and-cdk.md

```diff
@@ -0,0 +1,210 @@
+    }))
+        .then(v => callback(null, v), callback);
+};
+```
+
+(The code can be found on [the repository](https://github.com/Falydoor/event-sourcing-dynamodb/blob/master/resources/lambda.js))
+
+# Deployment and test
+
+## Deployment using CDK
+
+[AWS Cloud Development Kit](https://aws.amazon.com/cdk/) is a very powerful framework that allows you to model and provision your cloud application resources using familiar programming languages. All our stack is configured/deployed using CDK. To keep it simple, we are using TypeScript as the programming language.
+
+The CDK code below is fairly detailed and will create all the components for our stack:
+
+```ts
+import * as cdk from '@aws-cdk/core';
+import * as sqs from '@aws-cdk/aws-sqs';
+import * as dynamodb from '@aws-cdk/aws-dynamodb';
+import * as lambda from '@aws-cdk/aws-lambda';
+import { SqsEventSource } from '@aws-cdk/aws-lambda-event-sources';
+
+export class EventSourcingDynamodbStack extends cdk.Stack {
+  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    // Create queue
+    const queue = new sqs.Queue(this, 'queue', {
+      queueName: 'event-queue'
+    });
+
+    // Create lambda
+    const fn = new lambda.Function(this, 'fn', {
+      code: new lambda.AssetCode('resources'),
+      handler: 'lambda.handler',
+      runtime: lambda.Runtime.NODEJS_12_X,
+      functionName: 'event-to-dynamodb'
+    });
+
+    // Add lambda source
+    fn.addEventSource(new SqsEventSource(queue));
+
+    // Create dynamodb table
+    const sortKey = { name: 'Timestamp', type: dynamodb.AttributeType.STRING };
+    const table = new dynamodb.Table(this, 'table', {
+      partitionKey: { name: 'RequestId', type: dynamodb.AttributeType.STRING },
+      sortKey: sortKey,
+      tableName: 'event'
+    });
+
+    // Create GSI
+    table.addGlobalSecondaryIndex({
+      indexName: 'Type-Index',
+      partitionKey: { name: 'Type', type: dynamodb.AttributeType.STRING },
+      sortKey: sortKey
+    });
+    table.addGlobalSecondaryIndex({
+      indexName: 'Source-Index',
+      partitionKey: { name: 'Source', type: dynamodb.AttributeType.STRING },
+      sortKey: sortKey
+    });
+
+    // Grant write to lambda
+    table.grantWriteData(fn);
+  }
+}
+```
+
+(The code can be found on [the repository](https://github.com/Falydoor/event-sourcing-dynamodb/blob/master/lib/event-sourcing-dynamodb-stack.ts))
+
+CDK significantly simplifys your stack definition/deployment - saving you from a big headache.
+
+The commands below will deploy the whole stack:
+
+```sh
+cdk synth
+cdk deploy
+```
+
+(make sure to have your `AWS CLI` configured)
+
+## Testing
+
+In order to test the whole flow, we would need a Gateway that publishes a message to our SQS queue. For that, we will use `AWS CLI` to publish a message to the queue. The message content can be edited [here](https://github.com/Falydoor/event-sourcing-dynamodb/blob/master/test/message.json).
```

Comment:
:o I was hoping to see the example with the API gateway, especially because it is showed in your diagram.
Why not add it?